lib/api.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * REST API.
+ */
+
+// Internal packages:
+const package = require('../package.json')
+,   sink = require('./sink')
+,   util = require('./util')
+,   validator = require('./validator')
+;
+
+const Sink = sink.Sink
+,   version = package.version
+;
+
+/**
+ * Build a JSON response, and send it to the client.
+ *
+ * @param {Array} err - errors.
+ * @param {Array} warn - warnings.
+ * @param {Array} inf - informative messages.
+ * @param {Object} res - Express HTTP response.
+ * @param {Object} metadata - dictionary with some found metadata.
+ */
+
+const buildAndSendResult = function(err, warn, inf, res, metadata) {
+
+    const wrapper = {
+        success: 0 === err.length
+    ,   errors: err
+    ,   warnings: warn
+    ,   info: inf
+    ,   metadata: metadata
+    };
+
+    res.status(wrapper.success ? 200 : 400).send(wrapper);
+
+};
+
+/**
+ * Handle an API request: parse method and parameters, handle common errors and call the validator.
+ *
+ * @param {Object} req - Express HTTP request.
+ * @param {Object} res - Express HTTP response.
+ */
+
+const processRequest = function(req, res) {
+
+    if ('/api/version' === req.path) {
+        res.status(200).send(version);
+    }
+    else if ('/api/metadata' === req.path || '/api/validate' === req.path) {
+        const validate = ('/api/validate' === req.path);
+        var v
+        ,   v2
+        ,   options
+        ,   options2
+        ,   errors
+        ,   errors2
+        ,   warnings
+        ,   warnings2
+        ,   info
+        ,   info2
+        ,   handler
+        ,   handler2
+        ;
+        try {
+            options = util.processParams(
+                req.query,
+                undefined,
+                {required: (validate ? ['profile'] : []), forbidden: ['document', 'source']}
+            );
+        }
+        catch (err) {
+            return res.status(400).send(err.toString());
```

Comment:
Since the API is expected to return a JSON, it might be better to send a JSON in case of errors too. Same comment for line 145.